### DISCUSSION TITLE
Issue with BrotliDecode when hosting on itch.io.

### DISCUSSION THREAD

**Discussion by jonathanmccarthy** [kniEngine/kni#1271] [General]
I have several games made with the kniEngine hosted on itch.io.

They have all stopped working as an error is thrown on this line in index.html:
28:        import { BrotliDecode } from './js/decode.min.js';

To reproduce go here:
https://jonathanmcc.itch.io/sceal

The error displayed in Chrome is: 
Uncaught (in promise) Corrupted padding bits

The failing projects have not been updated on itch.io for about a year so something has changed externally.
Error also occurs in Firefox and Edge.
Error also occurs with a barebones solution made from the kni template with no changes.

This looks like the issue:
https://github.com/dotnet/aspnetcore/issues/45191

**Comment by jonathanmccarthy** [comment]
I tried using the unminified latest version of decode.js and that also has the issue.
https://github.com/google/brotli/blob/master/js/decode.js

![image](https://github.com/kniEngine/kni/assets/25285493/989269a0-7862-40fa-b029-d5d2a0b95b64)

**Comment by nkast** [comment]
We are following the issue here https://github.com/itchio/itch.io/issues/1512
The itch.io server or some proxy they have, is now uncompressing files on-the-fly and send the uncompressed content.

Meanwhile, as a workaround you can update this code from index.html
and set enableBrotliDecompression = false;
https://github.com/nkast/WebGLxnaProj/blob/main/WebGLxna/wwwroot/index.html#L32-L58

The drawback is that now it uses the uncompressed .DLLs , but there's is nothing we can do until it's fixed on itch.io.

**Reply by jonathanmccarthy** [reply]
Thanks for the rapid response. I applied the fix to the barebones solution I made and it worked.
There are new issues with my actual game (loading dlls such as newtonsoft.json etc). 

I'll investigate those.

**Comment by Vanglant** [comment]
Hello, had the same issue when deploying a KNI game to itch, and maybe this workaround could be useful for anyone else that faces the problem:

Rename the output .br files to something that Itch does not detect as .br, for example I added .data to it.
```csharp
<Target Name="RenameForItch" AfterTargets="Publish">
 <ItemGroup>
	<BrotliFiles Include="$(PublishDir)wwwroot\_framework\*.br" />
 </ItemGroup>
  <Copy
    SourceFiles="@(BrotliFiles)"
    DestinationFiles="@(BrotliFiles->'%(RootDir)%(Directory)%(Filename)%(Extension).data')"
  />
  <Delete Files="@(BrotliFiles)" />
</Target>
```

Then loading the wasm files like this in index.html, so that we use .br.data in itch:

```js
 const isItch = location.hostname.endsWith('.itch.io') || location.hostname.endsWith('.itch.zone');
....
 const response = await fetch(defaultUri + (isItch ? '.br.data': '.br'), { cache: 'no-cache' });
```

As you can see it is retrieving the correct file and the game works fine.
<img width="1315" height="42" alt="image" src="https://github.com/user-attachments/assets/261bb382-c671-471a-aad1-4fb4c436bd96" />
<img width="747" height="85" alt="image" src="https://github.com/user-attachments/assets/9d1b421b-2323-427d-8d97-6955c724ae09" />

Have not had luck with the .js files though, seems like Itch is actually compressing those on upload to the server not only by filename, but also by detecting file content.
